refactor(transformer/class-properties): use `duplicate_object` in `transform_expression_to_wrap_nullish_check`

Dunqing opened the pull request:
The `duplicate_object` is great, it has handled unbound identifiers and `this` expressions, so we can remove a lot of code. But it is still a little bit annoying because we need two `object` here,  although we can use `clone_in` it needs a special logic for `Expression::Identifier`.

overlookmotel reviewed (approved):
Rebased on main and made a couple of small tweaks.

overlookmotel commented:
### Merge activity

* **Dec 5, 12:41 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
refactor(transformer/class-properties): use `duplicate_object` in `transform_expression_to_wrap_nullish_check` (#7664)

The `duplicate_object` is great, it has handled unbound identifiers and `this` expressions, so we can remove a lot of code. But it is still a little bit annoying because we need two `object` here,  although we can use `clone_in` it needs a special logic for `Expression::Identifier`.
